### PR TITLE
widget and folder selector improvements

### PR DESCRIPTION
**Pull request by SIRSteiner**

### CODE REVIEW

**Comment by SIRSteiner** [comment]
IMG_20230218_215421
Smallest widget and a bigger

IMG_20230218_215641
New icon for none folder

File: app/src/main/res/values-de/strings.xml [hunk 1]
```diff
@@ -13,7 +13,7 @@
     <string name="link_url">Link-URL</string>
     <string name="pinned">Anpinnen</string>
     <string name="folder_name">Ordnername</string>
-    <string name="none">Ohne</string>
+    <string name="none">- keiner -</string>
```
Comment (AmrDeveloper) [inline app/src/main/res/values-de/strings.xml:16]: I think it butter to not using - - so to be like other languages

File: app/src/main/res/layout/widget_item_link.xml [hunk 2]
```diff
@@ -5,36 +5,40 @@
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
     android:padding="@dimen/dimen2dp"
+    android:theme="@style/ThemeOverlay.LinkHub.AppWidgetContainer"
```
Comment (AmrDeveloper) [inline app/src/main/res/layout/widget_item_link.xml:8]: For now it better to keep the background white to be similar to app white theme, maybe then we can try to support multi themes

**Review by AmrDeveloper — CHANGES_REQUESTED** [review]

**Comment by AmrDeveloper** [comment]
Note that it will be much easier for your to make each PR on a new branch so you can submit each one without waiting for the other to be merged for example

File: app/src/main/res/drawable/ic_directory_none.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (AmrDeveloper) [inline app/src/main/res/drawable/ic_directory_none.xml:1]: Nice idea for none folder icon

File: app/src/main/res/values-de/strings.xml (hunk 1 above)
Comment (SIRSteiner) [reply app/src/main/res/values-de/strings.xml:16]: I will change it with the next commit

File: app/src/main/res/layout/widget_item_link.xml (hunk 2 above)
Comment (SIRSteiner) [reply app/src/main/res/layout/widget_item_link.xml:8]: At the moment (latest official release) background depends on system design and also with that commit, that's not changed. But the (theme) text color for the widget in dark mode is only applying with that line.
IMG_20230219_103339
System light mode with commit

You can see light mode works at the widget.

Comment (AmrDeveloper) [reply app/src/main/res/layout/widget_item_link.xml:8]: You are right, great work 👍🏻

File: app/src/main/java/com/amrdeveloper/linkhub/data/Folder.kt [hunk 4]
```diff
@@ -15,7 +15,7 @@ data class Folder(
     @ColumnInfo(name = "name") var name: String,
     @ColumnInfo(name = "pinned") var isPinned: Boolean = false,
     @ColumnInfo(name = "click_count") var clickedCount: Int = 0,
-    @ColumnInfo(name = "color_name", defaultValue = "BLUE") var folderColor: FolderColor = FolderColor.BLUE,
+    @ColumnInfo(name = "color_name", defaultValue = "BLUE") var folderColor: FolderColor = FolderColor.NONE,
```
Comment (AmrDeveloper) [inline app/src/main/java/com/amrdeveloper/linkhub/data/Folder.kt:18]: We need to change the default value in column info to match the Folder NONE so it should be `NONE`

**Review by AmrDeveloper — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/amrdeveloper/linkhub/data/Folder.kt (hunk 4 above)
Comment (SIRSteiner) [reply app/src/main/java/com/amrdeveloper/linkhub/data/Folder.kt:18]: I don't know why but when I change this as suggested the app doesn't show any links nor folder and you can't add any link/folder.

**Comment by SIRSteiner** [comment]
I splitted these commits and made separate push requests as advised

**Comment by AmrDeveloper** [comment]
@SIRSteiner The problem is that we changed the DB Schema because we changed defaultValue I will check other PR first then we can think of solution or removing default value if it un needed

**Pull request closed without merging**